packages/widgets-manager/src/reducers/user.ts

```diff
@@ -0,0 +1,33 @@
+import { Action } from "../store/types";
+import { ActionType } from "../store/actions";
+import { App } from "../types";
+
+export const initialState = false;
+
+const updateUserDataReducer: any = (
+  _previousState: any = initialState,
+  action: Action<any>// action.payload = app object
+): App[] => {
+  return action.payload;
+};
+
+/**
+ * The User Reducer
+ * @param state: current application state
+ * @param action: the current action fired in the application
+ * @returns state: the new application state
+ */
+const reducer: any = (
+  prevState: any = initialState,
+  action: Action
+): any => {
+  switch (action.type) {
+    case ActionType.UPDATE_USER_DATA:
```

Comment:
When you update user data and save it to storage ? What about using the same library that uses widget but configure it with fire base?

Reply:
I thought about that and event investigated a little, and found out that to make redux-persist work with firebase there is a need to use react-redux-firebase library for bindings. Using it directly without React is not possible. So because widget-manager is plain vanilla JS I did not want to add react to it just to use redux-persist.

I thought about it and maybe in the future, if this project grows a lot we may want to add a lightweight version of React there are some options and so use it (https://github.com/preactjs/preact)
For now, we just do it without libraries.

Reply:
Created an issue for that: https://github.com/lets-talk/chat-utils/issues/69